spynnaker/pyNN/models/neuron/abstract_population_vertex.py

```diff
@@ -571,8 +755,557 @@ def reset_to_first_timestep(self):
+                    addr)
+                addr += size
+        return addr
+
+    def get_pop_table_size(self):
+        """ Get the size of the master population table in bytes
+
+        :rtype: int
+        """
+        return MasterPopTableAsBinarySearch.get_master_population_table_size(
+            self.__incoming_projections)
+
+    def get_synapse_expander_size(self):
+        """ Get the size of the synapse expander region in bytes
+
+        :rtype: int
+        """
+        size = 0
+        for proj in self.__incoming_projections:
+            synapse_info = proj._synapse_information
+            app_edge = proj._projection_edge
+            n_sub_edges = len(app_edge.pre_vertex.machine_vertices)
+            if not n_sub_edges:
+                vertex = app_edge.pre_vertex
+                max_atoms = float(min(vertex.get_max_atoms_per_core(),
+                                      vertex.n_atoms))
+                n_sub_edges = int(math.ceil(vertex.n_atoms / max_atoms))
+            size += self.__generator_info_size(synapse_info) * n_sub_edges
+
+        # If anything generates data, also add some base information
+        if size:
+            size += SYNAPSES_BASE_GENERATOR_SDRAM_USAGE_IN_BYTES
+            size += self.__neuron_impl.get_n_synapse_types() * BYTES_PER_WORD
+        return size
+
+    @staticmethod
+    def __generator_info_size(synapse_info):
+        """ The number of bytes required by the generator information
+
+        :rtype: int
+        """
+        if not synapse_info.may_generate_on_machine():
+            return 0
+
+        connector = synapse_info.connector
+        dynamics = synapse_info.synapse_dynamics
+        gen_size = sum((
+            GeneratorData.BASE_SIZE,
+            connector.gen_delay_params_size_in_bytes(synapse_info.delays),
+            connector.gen_weight_params_size_in_bytes(synapse_info.weights),
+            connector.gen_connector_params_size_in_bytes,
+            dynamics.gen_matrix_params_size_in_bytes
+        ))
+        return gen_size
+
+    @property
+    def synapse_executable_suffix(self):
+        """ The suffix of the executable name due to the type of synapses \
+            in use.
+
+        :rtype: str
+        """
+        if self.__synapse_dynamics is None:
+            return ""
+        return self.__synapse_dynamics.get_vertex_executable_suffix()
+
+    @property
+    def neuron_recordables(self):
+        """ Get the names of variables that can be recorded by the neuron
+
+        :rtype: list(str)
+        """
+        return self.__neuron_recorder.get_recordable_variables()
+
+    @property
+    def synapse_recordables(self):
+        """ Get the names of variables that can be recorded by the synapses
+
+        :rtype: list(str)
+        """
+        return self.__synapse_recorder.get_recordable_variables()
+
+    def get_common_constant_sdram(self, n_record, n_provenance):
```

Comment:
everything below this is really NOT meant to be in a APP vertex. But in the given splitter. coz the APP vertex has no authority to state that the synapse core's sdram usage is x. thats up to the splitter whom is making those machine verts.

Reply:
This is a utility method to avoid repetition in all the splitters.  I understand the concern, but there is nothing that makes a splitter call this if it wants to do something different.